### PR TITLE
[wasm-mt] Fixup C exports initialization after startup.ts refactoring

### PR DESCRIPTION
**Pull request by lambdageek**
Fixup issues due to d7c8bc10ac00fe330daa633f8a2b16ba208e6e56:

1. Add back the missing diagnostic server C wraps.
2. Run `init_c_exports` lazily on pthread workers, always.  Eager initialization doesn't work because at the time that our code runs, Emscripten hasn't fully initialized `Module` yet (in particular `Module["cwraps"]` hasn't been assigned yet).
3. Call `mono_wasm_pthread_worker_init` directly from `initializeImportsAndExports` - calling it from `preInit` doesn't work because Emscripten doesn't run `preInit` in pthread workers.  Also there's a technical change here to return a `Promise<EmscriptenModule>` - Emscripten is good with this because they use `.then` in `worker.js` and it's good for us because we can do async initialization at this point (for example waiting for the channel to the main browser thread to get established)

### CODE REVIEW

File: src/mono/mono/eventpipe/ep-rt-mono.c [hunk 1]
```diff
@@ -3784,7 +3784,17 @@ get_module_event_data (
...
+		if (image && image->filename) {
+			/* if there's a filename, use it */
+			module_data->module_il_path = image->filename;
+		} else if (image && image->module_name) {
+			/* otherwise, use the module name */
+			module_data->module_il_path = image->module_name;
+		}
+		if (!module_data->module_il_path)
+			module_data->module_il_path = "";
+
```
Comment (lambdageek) [inline src/mono/mono/eventpipe/ep-rt-mono.c:3797]: This is from #73018 - ignore it here.  it's not related (except that I need it for validation that the DS stuff is working as expected)

File: src/mono/wasm/runtime/startup.ts [hunk 2]
```diff
@@ -928,11 +924,18 @@ export function mono_wasm_set_main_args(name: string, allRuntimeArguments: strin
 /// 1. Emscripten skips a lot of initialization on the pthread workers, Module may not have everything you expect.
 /// 2. Emscripten does not run the preInit or preRun functions in the workers.
 /// 3. At the point when this executes there is no pthread assigned to the worker yet.
-async function mono_wasm_pthread_worker_init(): Promise<void> {
+export async function mono_wasm_pthread_worker_init(): Promise<void> {
+    console.debug("MONO_WASM: worker initializing essential C exports and APIs");
+    // FIXME: copy/pasted from mono_wasm_pre_init_essential - can we share this code? Any other global state that needs initialization?
+    init_c_exports();
+    cwraps_internal(INTERNAL);
```
Comment (pavelsavara) [inline src/mono/wasm/runtime/startup.ts:931]:
```cwraps_internal(INTERNAL);
cwraps_mono_api(MONO);
cwraps_binding_api(BINDING);
```
should be omitted I think. We are not having public API on worker any time soon.
And when we do, it's not these legacy methods.

**Review by pavelsavara — APPROVED** [review]

**Comment by lambdageek** [comment]
This isn't working. Still getting the assertion. I guess the stack trace could be misleading due to inlining.  Will try to repro locally

**Comment by lambdageek** [comment]
oops, I meant to do that to my other PR.  this one is ok.

**Pull request merged**